sobel/main.c

```diff
@@ -0,0 +1,143 @@
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <pthread.h>
+
+extern void* sobel(void* args);
+
+// struct for transporting data between threads
+struct data {
+    int width;
+    int start;
+    int end;
+    char** base;
+    char** res;
+};
+
+int main(int argc, char* argv[]) {
+    if (argc != 2) {
+        printf("No input file or too much arguments\n");
+        exit(1); 
+    }
+    
+    char* filename = argv[1];
+    FILE* input;
+    FILE* output;
+    
+    // checking the file
+    if ((input = fopen(filename, "r")) == NULL) {
+        printf("Wrong filename");
+        exit(1);
+    }
+    
+    char* info = (char*)malloc(54);
```

Comment:
In C there is no need to explicit cast for malloc, it returns void *, which automatically converted to correct type. It relates to all variables in this PR.
```suggestion
    char* info = malloc(54);
```